core/src/main/java/org/jdbi/v3/core/Enums.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jdbi.v3.core;
+
+import java.lang.annotation.Annotation;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.jdbi.v3.core.config.JdbiConfig;
+import org.jdbi.v3.core.internal.AnnotationFactory;
+import org.jdbi.v3.core.qualifier.QualifiedType;
+import org.jdbi.v3.core.qualifier.Qualifiers;
+
+/**
+ * Configuration for behavior related to {@link Enum}s.
+ */
+public class Enums implements JdbiConfig<Enums> {
+    private EnumHandling handling;
+
+    public Enums() {
+        handling = EnumHandling.BY_NAME;
+    }
+
+    private Enums(Enums other) {
+        handling = other.handling;
+    }
+
+    public EnumHandling getDefaultHandling() {
+        return handling;
+    }
+
+    public Enums defaultByName() {
+        this.handling = EnumHandling.BY_NAME;
+        return this;
+    }
+
+    public Enums defaultByOrdinal() {
+        this.handling = EnumHandling.BY_ORDINAL;
+        return this;
+    }
+
+    public <E extends Enum<E>> EnumHandling findStrategy(QualifiedType<E> givenType, Class<E> clazz) {
+        return deriveStrategy(givenType)
+            .orElseGet(() -> deriveStrategy(qualifyFromSource(clazz))
+                .orElse(handling));
+    }
+
+    private <E extends Enum<E>> Optional<EnumHandling> deriveStrategy(QualifiedType<E> type) {
+        boolean hasByName = type.hasQualifier(EnumByName.class);
+        boolean hasByOrdinal = type.hasQualifier(EnumByOrdinal.class);
+
+        if (hasByName && hasByOrdinal) {
```

Comment:
This is an advantage this approach has over @stevenschlansker 's. The latter can check for this conflict when adding qualifiers from source, but not when the user builds the qualifiedtype themself because the factories are matched by qualifiers and registration order.